### ISSUE TITLE
change trigger zone t otrigger box with visibility and scale

### ISSUE THREAD

**Issue by JCLEONITA** [TheGameCreators/GameGuruRepo#4880]
see it like an enanchement  but  it should be more flexible and easy t ouse having a visual parameter t omanaged it
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/55716993/fe9af58f-0ac5-44dd-b388-74af13604b73)

**Comment by MonkeyFrogStudio** [comment]
I agree. I really don't like the way we have to work with any of the zones in MAX. They are a bit of a pain. To be able to visually see all three dimensions as you adjust things would be great. I'd also made this recommendation for zones some time back:

https://github.com/TheGameCreators/GameGuruRepo/issues/4362

**Comment by synchromesh62** [comment]
I would also love to see something like this but zones have 5 points and the points are removable.
It does look nice as a cube but adjusting all the points etc i can see it being more awkward unless you can toggle it on and off.
The zones need sorting anyway just to keep them on the ground.

**Comment by MonkeyFrogStudio** [comment]
It would be good, I think, if when a zone is selected, there would be a place on the right-side panel to enter in a number of sides. That way, if you want to have a zone with 3, 4, 6, 8, etc., sides, you'd just enter the number and the zone would change shape. Then you could move the different points about with the gizmo (see my previous post with link) to shape the zone as you please.

**Comment by Necrym59** [comment]
A nice visual feature, i think this is a good idea to a certain degree, extra points on zones can be done by dbl clicking on the corner joins now but a sides panel would be another option. To be honest i think spline drawing of any zones would be a much  better overall solution anyway, and once a zone is drawn then if the option is selected, then creates the 3d visual transparent walls to the zone height in the editor.

**Comment by TenebreGaming** [comment]
This could easily be done through scripting and a simple cube entity, and it wouldn't take too much extra compute power over it being done internally (unless the GPU or SIMD instructions are used to perform all of the checks at once, which would made iternal much faster). Of course, if a lot of zones are needed, internal would definitely be preferred, but I don't think that even a 5kx5k level would need so many zones that a script would cause lag.

**Comment by MonkeyFrogStudio** [comment]
There's already a behavior (or at least I have one) that can do this. But that's not the point. The point is that MAX has trigger zones, but they are not visually the best. They don't visually show the height of the zone, for example. Also, it's a bit of a pain to resize/reshape them. So, instead of someone creating a script that requires a cube entity, it would be a heck of a lot nicer to have the current zones in MAX redone to be more user-friendly and powerful.

**Comment by JCLEONITA** [comment]
> There's already a behavior (or at least I have one) that can do this. But that's not the point. The point is that MAX has trigger zones, but they are not visually the best. They don't visually show the height of the zone, for example. Also, it's a bit of a pain to resize/reshape them. So, instead of someone creating a script that requires a cube entity, it would be a heck of a lot nicer to have the current zones in MAX redone to be more user-friendly and powerful.

totally agree

**Comment by TenebreGaming** [comment]
Yeah, I'm not disagreeing that it would be great it MAX could do it. It would definitely make it better!
